projects/packages/backup/src/class-rest-controller.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * The Backup Rest Controller class.
+ * Registers the REST routes for Backup.
+ *
+ * @package automattic/jetpack-backup
+ */
+
+namespace Automattic\Jetpack\Backup;
+
+use Automattic\Jetpack\Connection\Rest_Authentication;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Server;
+
+/**
+ * Registers the REST routes for Backup.
+ */
+class REST_Controller {
+	/**
+	 * Registers the REST routes for Backup.
+	 *
+	 * @access public
+	 * @static
+	 */
+	public static function register_rest_routes() {
+		// Install a Helper Script to assist Jetpack Backup fetch data.
+		register_rest_route(
+			'jetpack/v4',
+			'/backup-helper-script',
+			array(
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => __CLASS__ . '::install_backup_helper_script',
+				'permission_callback' => __CLASS__ . '::backup_helper_script_permissions_callback',
+				'args'                => array(
+					'helper' => array(
+						'description' => __( 'base64 encoded Backup Helper Script body.', 'jetpack' ),
+						'type'        => 'string',
+						'required'    => true,
+					),
+				),
+			)
+		);
+
+		// Delete a Backup Helper Script.
+		register_rest_route(
+			'jetpack/v4',
+			'/delete-backup-helper-script',
```

Comment:
I wanted a more RESTfull approach here, aka:
Install Helper Script: `POST /backup-helper-script`
Delete Helper Script: `DELETE /backup-helper-script`

However Jetpack Connection [restricts us](https://github.com/Automattic/jetpack/blob/master/projects/packages/connection/src/class-rest-authentication.php#L133) from using any HTTP methods other than `GET` or `POST`.

I also tried to call the `delete` endpoint using `POST` and `_method=DELETE` but that leaves us with 2 exactly same endpoint definitions. Am I missing something here? Any suggestion is more than welcome.

Reply:
If you send a POST request with `_method=DELETE`, wouldn't you have two endpoints, one with `'methods' => WP_REST_Server::CREATABLE` and one with `'methods' => WP_REST_Server::DELETABLE`?

Reply:
Right! For some reason it was always failing when I was testing, with `rest_route_not_found`.  This made me wonder whether I should accept both `POST` and `DELETE` methods in the endpoint definition, which didn't sound right though.
I tried once again today using `X-HTTP-Method-Override` and worked so I modified the PR and the one bin script accordingly: 29958-pb/#php